language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/contexts/AbstractDocumentServiceContext.java

```diff
@@ -53,11 +53,11 @@
  */
 public class AbstractDocumentServiceContext implements DocumentServiceContext {
 
-    private final LSOperation operation;
+    private LSOperation operation;
```

Comment:
Is there a reason to remove the final keyword?

Reply:
if not, we have to add it as a constructor parameter.